packages/components/src/List/Toolbar/Toolbar.component.js

```diff
@@ -81,60 +79,47 @@ function Toolbar({
 
 	return (
-		<div className="tc-list-toolbar">
+		<div className={classNames(theme['tc-list-toolbar'], 'tc-list-toolbar')}>
 			{injected('before-actionbar')}
-			{actionBar && <Renderer.ActionBar {...actionBarProps} />}
-			{injected('after-actionbar')}
-			{injected('before-navbar')}
-			{hasToolbarItem && (
-				<Navbar componentClass="div" className={theme['tc-list-toolbar']} role="toolbar" fluid>
-					{injected('before-selectall')}
-					{selectAllCheckbox && <SelectAll {...selectAllCheckbox} t={t} />}
-					{injected('after-selectall')}
-					{injected('before-displaymode')}
-					{display && (
-						<Label
-							text={t('LIST_TOOLBAR_DISPLAY', { defaultValue: 'Display:' })}
-							htmlFor={displayModeId}
-						/>
-					)}
-					{display && <SelectDisplayMode id={displayModeId} {...display} t={t} />}
-					{injected('after-displaymode')}
-					{injected('before-sort')}
-					{sort && (
-						<Label
-							text={t('LIST_TOOLBAR_SORT_BY', { defaultValue: 'Sort by:' })}
-							htmlFor={id && `${id}-sort-by`}
-						/>
-					)}
-					{sort && <SelectSortBy id={id && `${id}-sort`} {...sort} t={t} />}
-					{injected('after-sort')}
-					{injected('before-pagination')}
-					{pagination && (
-						<Label
-							text={t('LIST_TOOLBAR_PAGINATION_SHOW', { defaultValue: 'Show:' })}
-							htmlFor={id && `${id}-pagination-size`}
-						/>
-					)}
-					{pagination && <Pagination id={id && `${id}-pagination`} {...pagination} t={t} />}
-					{injected('after-pagination')}
-					{injected('before-filter')}
-					{filter && (
-						<Renderer.FilterBar
-							id={id && `${id}-filter`}
-							{...filter}
-							t={t}
-							navbar
-							className="navbar-right"
-						/>
-					)}
-					{injected('after-filter')}
-				</Navbar>
+			{actionBar && (
+				<Renderer.ActionBar {...actionBarProps}>
+					<ActionBar.Content right>
+						<ul>
+							{injected('before-filter')}
+							{filter && (
+								<li className="separated">
+									<Renderer.FilterBar id={id && `${id}-filter`} {...filter} t={t} navbar />
+								</li>
+							)}
+							{injected('after-filter')}
+							{injected('before-sort')}
+							{showSort && (
+								<li className="select-sort-by separated">
+									<Label
+										text={t('LIST_TOOLBAR_SORT_BY', { defaultValue: 'Sort by:' })}
+										htmlFor={id && `${id}-sort-by`}
+									/>
+									<SelectSortBy id={id && `${id}-sort`} {...sort} t={t} />
+								</li>
+							)}
+							{injected('after-sort')}
+							{injected('before-displaymode')}
+							{display && (
+								<li className="separated">
+									<DisplayModeToggle id={displayModeId} {...display} t={t} />
+								</li>
+							)}
+							{injected('after-displaymode')}
+						</ul>
+					</ActionBar.Content>
+				</Renderer.ActionBar>
 			)}
-			{injected('after-navbar')}
+			{injected('after-actionbar')}
+			{showSelectAll && <SelectAll {...selectAllCheckbox} t={t} />}
```

Comment:
This is not in the toolbar, it should not be rendered here.
I think List.component.js > ListToolbar is  a better place.